scheduler/test/cook/test/rest/api.clj

```diff
@@ -709,37 +716,41 @@
                                   {:request-method :post
                                    :body-params {:user "foo"
                                                  :quota new-quota
-                                                 :reason "Needs custom settings"}}))
+                                                 :reason "Needs custom settings"
+                                                 :pool "test-pool"}}))
             update-body (response->body-data update-resp)
             _ (is (<= 200 (:status update-resp) 299))
             _ (is (= (kw-keys update-body) new-quota))
             get-resp (h (merge quota-req-attrs
                                {:request-method :get
                                 :query-params {:user "foo"}}))
             get-body (response->body-data get-resp)]
-        (is (= get-body update-body))))
-
-    (testing "gpu quota is checked on job submission"
-      (let [job (assoc (basic-job) "gpus" 3.0)
-            job-resp (h (merge quota-req-attrs
-                               {:uri "/rawscheduler"
-                                :authorization/user "foo"
-                                :request-method :post
-                                :body-params {"jobs" [job]}}))]
-        (is (<= 200 (:status job-resp) 299)))
-      (let [job (assoc (basic-job) "gpus" 4.0)
-            job-resp (h (merge quota-req-attrs
-                               {:uri "/rawscheduler"
-                                :authorization/user "foo"
-                                :request-method :post
-                                :body-params {"jobs" [job]}}))]
-        (is (= 422 (:status job-resp)))))
-
+        (is (= (get-in get-body ["pools" "test-pool"]) update-body))))
+
+    (with-redefs [config/valid-gpu-models (constantly [{:pool-regex "test-pool"
+                                                        :valid-models #{"nvidia-tesla-p100"}
+                                                        :default-model "nvidia-tesla-p100"}])]
+      (testing "gpu quota is checked on job submission"
+        (let [job (assoc (basic-job) "gpus" 3.0)
```

Comment:
Is GPU's as an argument supposed to be a float or int. I'm a bit surprised to see float here. What if someone sends 1.5 GPU's? (Do we have a test for this?)

Reply:
it accepts floats that are whole numbers or int. added a test in gpus-api requesting 1.5 gpus and the job submission fails.

Reply:
there is inconsistency in the rest of the file for float or int but I modified this test to use ints.

Reply:
changing gpus from 3.0 to 3 in this test causes failures

Reply:
quota endpoint lets a user set the gpus quota to a float, and even if you set gpus as an int, it returns it as a float